### ISSUE TITLE
Feature Request: Add support for handling browser forward/back buttons

### ISSUE THREAD

**Issue by greenlaw** [awslabs/aws-js-s3-explorer#90]
I've noticed that the object-key hash-navigation only seems to work when the breadcrumbs are clicked or the user navigates to the page with a populated hash (e.g. initial load or browser refresh).  If you're navigating through a key (directory) hierarchy, the browser back/forward buttons don't seem to have any effect, even though the location hash changes. 

Looks like this could be implemented by adding a handler for the [popstate event](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event).

When I find some time hopefully I can create a PR for this, but wanted to document it before I forgot.

**Comment by lpalbou** [comment]
Yep, I confirm that would be helpful, many users still use that to go through different directories.
